nhentai/command.py

```diff
@@ -58,9 +58,10 @@ def main():
 
         if options.is_save_download_history:
             with DB() as db:
-                data = set(db.get_all())
-
-            doujinshi_ids = list(set(doujinshi_ids) - data)
+                data = db.get_all()
+                for i in range(0, len(data)):
+                    data[i] = int(data[i])
+            doujinshi_ids = [x for x in doujinshi_ids if x not in set(data)]
```

Comment:
**doujinshi_ids** where not substracting elements from **data** because at first _list - list can't be possible_, but with [List Comprehensions](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)  and transform **string list from data** to **int list from data** :
```
for i in range(0, len(data)):
     data[i] = int(data[i])
```